python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -387,31 +388,55 @@ class RandomForestClassifier(Base):
                              " please read the documentation")
 
     def _obtain_treelite_handle(self):
-        task_category = CLASSIFICATION_MODEL
+        cdef ModelHandle cuml_model_ptr = NULL
+        cdef RandomForestMetaData[float, int] *rf_forest = \
+            <RandomForestMetaData[float, int]*><size_t> self.rf_forest
         if self.num_classes > 2:
             raise NotImplementedError("Pickling for multi-class "
                                       "classification models is currently not "
                                       "implemented. Please check cuml issue "
                                       "#1679 for more information.")
+        if self.treelite_handle is None:
+            task_category = CLASSIFICATION_MODEL
+            build_treelite_forest(
+                & cuml_model_ptr,
+                rf_forest,
+                <int> self.n_cols,
+                <int> task_category,
+                <vector[unsigned char] &> self.model_pbuf_bytes)
```

Comment:
Instead of casting to `vector[unsigned char]`, which goes through a slower Cython path that [iterates over a generic Python object]( https://github.com/cython/cython/blob/00c1dc96af0e44741dc777f96a2eb5769f572bb3/Cython/Utility/CppConvert.pyx#L45-L49 ), would suggest leveraging `std::vector`'s `assign` method with a Cython `memoryview` so that C++ can allocate a block of memory and do a single `memcpy`. Should be more performant.

```suggestion
        cdef unsigned char[::1] model_pbuf_mv = self.model_pbuf_bytes
        cdef vector[unsigned char] model_pbuf_vec
        with cython.boundscheck(False):
            model_pbuf_vec.assign(&model_pbuf_mv[0], & model_pbuf_mv[model_pbuf_mv.shape[0]])
        if self.treelite_handle is None:
            task_category = CLASSIFICATION_MODEL
            build_treelite_forest(
                & cuml_model_ptr,
                rf_forest,
                <int> self.n_cols,
                <int> task_category,
                model_pbuf_vec)
```